model_configs/multilabel_classification/v2.5/bgc_tasknn.jsonnet

```diff
@@ -0,0 +1,155 @@
+// Simple tasknn only model for blurb genre collection
+local test = std.extVar('TEST');  // a test run with small dataset
+local data_dir = std.extVar('DATA_DIR');
+local cuda_device = std.extVar('CUDA_DEVICE');
+local use_wandb = (if test == '1' then false else true);
+
+local dataset_name = 'bgc';
+local dataset_metadata = (import '../datasets.jsonnet')[dataset_name];
+local num_labels = dataset_metadata.num_labels;
+local num_input_features = dataset_metadata.input_features;
+
+// model variables
+local ff_hidden = std.parseJson(std.extVar('ff_hidden'));
+local label_space_dim = ff_hidden;
+local task_nn_dropout = std.parseJson(std.extVar('task_nn_dropout_10x')) / 10.0;
+local ff_activation = 'softplus';
+local ff_linear_layers = std.parseJson(std.extVar('ff_linear_layers'));
+local task_nn_weight_decay = std.parseJson(std.extVar('task_nn_weight_decay'));
+local gain = (if ff_activation == 'tanh' then 5 / 3 else 1);
+local transformer_model = 'bert-base-uncased';  // huggingface name of the model
+local transformer_dim = 768;
+{
+  [if use_wandb then 'type']: 'train_test_log_to_wandb',
+  evaluate_on_test: true,
+  // Data
+  dataset_reader: {
+    type: 'bgc',
+    [if test == '1' then 'max_instances']: 100,
+    token_indexers: {
+      x: {
+        type: 'pretrained_transformer',
+        model_name: transformer_model,
+      },
+    },
+    tokenizer: {
+      type: 'pretrained_transformer',
+      model_name: transformer_model,
+      max_length: 512,
+    },
```

Comment:
Text based dataset readers take token_indexers and tokenizers. See AllenNLP Guide to understand their roles.